src/main/java/io/vertx/core/net/impl/SSLHelper.java

```diff
@@ -291,18 +319,26 @@ private Future<EngineConfig> build(SSLOptions sslOptions, ContextInternal ctx) {
       }).compose(v2 -> ctx.<EngineConfig>executeBlockingInternal(p -> {
         Supplier<SslContextFactory> supplier;
         boolean useWorkerPool;
+        SslProvider sslProvider;
         try {
           SSLEngineOptions resolvedEngineOptions = resolveEngineOptions(sslEngineOptions, useAlpn);
           supplier = resolvedEngineOptions::sslContextFactory;
           useWorkerPool = resolvedEngineOptions.getUseWorkerThread();
+          if (resolvedEngineOptions instanceof JdkSSLEngineOptions) {
```

Comment:
This would need a substitution in native. Can we move this to a separated, easily substituable method.

Reply:
I've now performed a single check on an already present option now  i.e. `JdkSSLEngineOptions`
and removed the one which is related OpenSSL, it looks better?
I'm using this additional time to add a further test, as suggested by @vietj